kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -205,6 +205,26 @@ public boolean hasKernelWriteSupport(Metadata metadata) {
     }
   }
 
+  public static final TableFeature VARIANT_SHREDDING_PREVIEW_RW_FEATURE =
```

Comment:
add unit tests in `TableFeaturesSuite.scala`. You can find examples there. Basically, test to make sure this config enables the feature, hasReadSupport and hasWriteSupport works as expected and few other cases (look at variantType tests)